internal/pkg/es/result.go

```diff
@@ -54,8 +57,63 @@ type HitsT struct {
 	MaxScore *float64 `json:"max_score"`
 }
 
+type Bucket struct {
+	Key          string           `json:"key"`
+	DocCount     int64            `json:"doc_count"`
+	Aggregations map[string]HitsT `json:"-"`
+}
+
+type _bucket Bucket
+
+func (b *Bucket) UnmarshalJSON(data []byte) error {
+	b2 := _bucket{}
+	err := json.Unmarshal(data, &b2)
+	if err != nil {
+		return err
+	}
+	var aggs map[string]interface{}
+	err = json.Unmarshal(data, &aggs)
+	if err != nil {
+		return err
+	}
+	typ := reflect.TypeOf(b2)
```

Comment:
can this be done without reflection?

Reply:
I will remove the reflection, but we need to be sure that any keys we add to the `Bucket` type, is also updated in the `UnmarshalJSON`. I added some comments in hopes that it is not missed.